cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go

```diff
@@ -632,168 +619,168 @@ func TestControllerNodeGroupForNodeWithPositiveScalingBounds(t *testing.T) {
 	}
 
 	t.Run("MachineSet", func(t *testing.T) {
-		testObjs := newMachineSetTestObjs(t.Name(), 0, 1, 1, map[string]string{
+		testConfig := mustCreateMachineSetTestConfig(testNamespace, 1, 1, map[string]string{
 			nodeGroupMinSizeAnnotationKey: "1",
 			nodeGroupMaxSizeAnnotationKey: "1",
 		})
-		test(t, testObjs)
+		test(t, testConfig)
 	})
 
 	t.Run("MachineDeployment", func(t *testing.T) {
-		testObjs := newMachineDeploymentTestObjs(t.Name(), 0, 1, 1, map[string]string{
+		testConfig := mustCreateMachineDeploymentTestConfig(testNamespace, 1, 1, map[string]string{
 			nodeGroupMinSizeAnnotationKey: "1",
 			nodeGroupMaxSizeAnnotationKey: "1",
 		})
-		test(t, testObjs)
+		test(t, testConfig)
 	})
 }
 
 func TestControllerNodeGroups(t *testing.T) {
-	type testCase struct {
-		description string
-		annotations map[string]string
-		errors      bool
-		nodegroups  int
+	assertNodegroupLen := func(t *testing.T, controller *machineController, expected int) {
+		t.Helper()
+		nodegroups, err := controller.nodeGroups()
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if l := len(nodegroups); l != expected {
+			t.Fatalf("expected %d, got %d", expected, l)
+		}
 	}
 
-	var testCases = []testCase{{
-		description: "errors with bad annotations",
-		annotations: map[string]string{
-			nodeGroupMinSizeAnnotationKey: "-1",
-			nodeGroupMaxSizeAnnotationKey: "10",
-		},
-		nodegroups: 0,
-		errors:     true,
-	}, {
-		description: "success with zero bounds",
-		nodegroups:  0,
-		errors:      false,
-	}, {
-		description: "success with positive bounds",
-		annotations: map[string]string{
-			nodeGroupMinSizeAnnotationKey: "1",
-			nodeGroupMaxSizeAnnotationKey: "10",
-		},
-		nodegroups: 1,
-		errors:     false,
-	}}
+	annotations := map[string]string{
+		nodeGroupMinSizeAnnotationKey: "1",
+		nodeGroupMaxSizeAnnotationKey: "2",
+	}
 
-	test := func(t *testing.T, tc testCase, testObjs *clusterTestConfig) {
-		controller, stop := testObjs.newMachineController(t)
-		defer stop()
+	controller, stop := mustCreateTestController(t)
+	defer stop()
 
-		nodegroups, err := controller.nodeGroups()
-		if tc.errors && err == nil {
-			t.Errorf("expected an error")
-		}
+	// Test #1: zero nodegroups
+	assertNodegroupLen(t, controller, 0)
 
-		if !tc.errors && err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
+	// Test #2: add 5 machineset-based nodegroups
+	machineSetConfigs := mustCreateMachineSetTestConfigs("MachineSet", 5, 1, 1, annotations)
```

Comment:
You are creating machineset with 5 nodes but annotations limits the max. number of nodes per node group to 2. Is that OK?

Reply:
> You are creating machineset with 5 nodes but annotations limits the max. number of nodes per node group to 2. Is that OK?

This one is perhaps a little subtle. Calling the plural version of `mustCreateMachineSetTestConfigs` creates multiple machine sets/deployments. Here w'ere asking for 5.